Source/WebCore/animation/AnimationTimelinesController.cpp

```diff
@@ -329,6 +334,18 @@ void AnimationTimelinesController::cacheCurrentTime(ReducedResolutionSeconds new
         CancellableTask task(m_pendingAnimationsProcessingTaskCancellationGroup, std::bind(&AnimationTimelinesController::processPendingAnimations, this));
         m_document->eventLoop().queueTask(TaskSource::InternalAsyncTask, WTFMove(task));
     }
+
+    if (!m_isSuspended) {
+        // In order to not have a stale cached current time, we schedule a timer to reset it
+        // in the time it would take an animation frame to run under normal circumstances.
+        ASSERT(m_document->page());
+        m_cachedCurrentTimeClearanceTimer.startOneShot(RefPtr { m_document->page() }->preferredRenderingUpdateInterval());
```

Comment:
Would it be better to explicitly clear the cached time in `Page::doAfterUpdateRendering()`? This seems like it could be flaky.

Reply:
No, the cached current time must remain cached until the next page rendering update.